### PR TITLE
core, mobile: add group feature to allow direct messages

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: src/Simplex/Chat.hs [hunk 1]
```diff
@@ -3132,11 +3132,11 @@ featureProhibited forWhom Contact {mergedPreferences} = \case
      in if forWhom enabled then Nothing else Just CFVoice
   _ -> Nothing
 
-groupFeatureProhibited :: GroupInfo -> MsgContent -> Maybe ChatFeature
+groupFeatureProhibited :: GroupInfo -> MsgContent -> Maybe GroupFeature
```
Comment (jr-simplex) [inline src/Simplex/Chat.hs:3135]: e should check "direct messages" feature when sending/processing new direct content message, somewhere around `featureProhibited` (not this function, direct one). I don't think relying only on UI is a good idea, especially considering we sometimes have bugs with state updates, etc., also as a sanity check in backend. And not even only just a sanity check actually - on group preference change sender may not receive this feature update (due to fragmentation w/ owner for example) - at least receiver will be "protected" and creating "rejected" items.

Comment (jr-simplex) [reply src/Simplex/Chat.hs:3135]: we should also add a test for it

Comment (jr-simplex) [reply src/Simplex/Chat.hs:3135]: the check can be based on the `directContact` function for example

Comment (epoberezkin) [reply src/Simplex/Chat.hs:3135]: yes, next pr

**Pull request merged**